### PR TITLE
Better generator

### PR DESCRIPTION
**Pull request by Jelly9912**

### CODE REVIEW

**Comment by matcracker** [comment]
Who can test this?

**Comment by thebigsmileXD** [comment]
Where does the code come from? The better generators plugin?

**Comment by Jelly9912** [comment]
Me

**Comment by Jelly9912** [comment]
Yes

**Comment by robske110** [comment]
I will test it...

File: src/pocketmine/level/generator/normal/Normal.php [hunk 1]
```diff
@@ -35,24 +35,28 @@
...
-
-
+use pocketmine\level\generator\populator\TallGrass;
+use pocketmine\level\generator\populator\Tree;
 use pocketmine\level\Level;
 use pocketmine\math\Vector3 as Vector3;
 use pocketmine\utils\Random;
 
 class Normal extends Generator{
+    const NAME="Normal";
```
Comment (thebigsmileXD) [inline src/pocketmine/level/generator/normal/Normal.php]: Fix the spacing here.

**Comment by Jelly9912** [comment]
screenshot_2015-12-25-18-21-40

**Comment by thebigsmileXD** [comment]
I think the code will work.

**Comment by Jelly9912** [comment]
Caves and mountains looks good

**Comment by Jelly9912** [comment]
screenshot_2015-12-25-18-16-31

**Comment by matcracker** [comment]
Why in the image there is unloaded chunks?

**Comment by thebigsmileXD** [comment]
I see a problem.

```php
  		$this->selector = new BiomeSelector($this->random, function($temperature, $rainfall){
  			if($rainfall < 0.25){
 -				if($rainfall < 0.7){
 +				if($temperature < 0.7){
```
Wether change $temperature back or change it in Biome selector. the first would be better

**Comment by thebigsmileXD** [comment]
> @matcracker Why in the image there is unloaded chunks?
Render distance

**Comment by matcracker** [comment]
Lol, render distance -5 xD

**Comment by thebigsmileXD** [comment]
No. the one picture is in spectator mode, the other has a distance of.. 16 i guess

**Comment by aodzip** [comment]
Seems good 
Really Good Generater

**Comment by matcracker** [comment]
True... I forgot the spectator mode

**Comment by thebigsmileXD** [comment]
The spacing and intending in the file really need a fix. And this should be pushed to the official PocketMine.

**Comment by Jelly9912** [comment]
Sounds good 👍

**Comment by aodzip** [comment]
Merge this?

**Comment by thebigsmileXD** [comment]
I still don't think the temperature and rainfall switch is correct.
Think about it..
Why should it be a river if the temperature is right, but shouldn't it depend on the rainfall?

**Comment by matcracker** [comment]
We can do $rainfall < 0.7 || $temperature < 0.7

**Comment by matcracker** [comment]
0.25 not 0.7 xD

**Comment by aodzip** [comment]
Merge this generator and improve it later?

**Pull request merged**